File: Software-Praktikum/src/server/bo/MessageBO.py
from .Businessobject import Businessobject;


class MessageBO (Businessobject):
    # Instanziierung des Objekts der Klasse Message
    def __init__(self):
        super().__init__()
        self.profile_id = None
        self.room = None
        self.text = None
#        self.time = None;

    def set_profile_id(self, key):
        self.profile_id = key

    def get_profile_id(self):
        return self.profile_id

    def set_room(self, room):
        self.room = room

    def get_room(self):
        return self.room

    def set_text(self, text):
        self.text = text

    def get_text(self):
        return self.text

#    def get_time(self):
#        return self.time;

#    def set_time(self):
#        this.time = datetime.now(tz = None);
     
    def __str__(self):
        return "Id: {}, Message von Profil {} im Chat {}: {}".format(self.get_id(), self.get_profile_id(),
                                                                     self.get_room(), self.get_text())

    # Umwandlung eines Dictonary Eintrags in ein Message Objekt
    @staticmethod
    def from_dict(dictionary=dict()):
        obj = MessageBO()
        obj.set_id(dictionary["id"])
        obj.set_profile_id(dictionary["profile_id"])
        obj.set_room(dictionary["room"])
        obj.set_text(dictionary["text"])
        return obj

File: Software-Praktikum/Frontend/lernapp/src/API/MessageBO.js
import BusinessObject from './BusinessObject';

/**
 * Die einzelnen Nachrichten eines Chatraums als BO
 * 
 * @author [Ha Mi Duong](https://github.com/HamiDuong)
 */
export default class MessageBO extends BusinessObject {

    /**
     * 
     * @param {int} profile_id - Id der Person welche die Nachricht geschrieben hat
     * @param {int} room - Raum in dem die MessageBO ist
     * @param {string} text - Text der Nachricht
     */
    constructor(profile_id, room, text){
        super();
        this.profile_id = profile_id;
        this.room = room;
        this.text = text;
    }

    //Getter und Setter Methoden für die Objekte der Klasse MessageBO
    set_profile_id(id){
        this.profile_id = id;
    }

    get_profile_id(){
        return this.profile_id;
    }

    set_room(id){
        this.room = id;
    }

    get_room(){
        return this.room;
    }

    set_text(text){
        this.text = text;
    }

    get_text(){
        return this.text;
    }

    static fromJSON(message){
        let result = [];

        if(Array.isArray(message)){
            message.forEach((c) => {
                Object.setPrototypeOf(c, MessageBO.prototype);
                result.push(c);
            })
        }else{
            let c = message;
            Object.setPrototypeOf(c, MessageBO.prototype);
            result.push(c);
        }
        return result;
    }
}